src/components/Search/SearchInputAndResults.tsx

```diff
@@ -68,9 +67,7 @@ export default function SearchInputAndResults({
 
   const searchRecentRevisions = useCallback(
     async (searchTerm: string) => {
-      const emailMatch = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
-      const longHashMatch = /\b[a-f0-9]{40}\b/;
-      const shortHashMatch = /\b[a-f0-9]{12}\b/;
+      const authorInfoMatch = /[^0-9a-fA-F]/;
```

Comment:
nit: please add a comment explaining the reasoning, something such as:
```
// If the searchTerm doesn't look like a hash, then we assume it might be an author.
// If it looks like a hash, then we assume it's a hash.
// NOTE: In the future we might want to be more clever and request both
// endpoints even when it looks like a hash. For example a request such as "ade" could
// return results for an author named "adenot" _and_  results for a hash "ade821ac".
// In that case it would be better to show the results for the author.
```